azurelinuxagent/common/protocol/imds.py

```diff
@@ -264,3 +268,52 @@ def get_compute(self):
         set_properties('compute', compute_info, data)
 
         return compute_info
+
+    def validate(self):
+        """
+        Determines whether the metadata instance api returns 200, and the response
+        is valid: compute should contain location, name, subscription id, and vm size
+        and network should contain mac address and private ip address.
+        :return: Tuple<is_healthy:bool, error_response:str>
+            is_healthy: True when validation succeeds, False otherwise
+            error_response: validation failure details to assist with debugging
+        """
+
+        # ensure we get a 200
+        resp = restutil.http_get(self.instance_url, headers=self._headers)
+        if restutil.request_failed(resp):
+            return False, "{0}".format(restutil.read_response_error(resp))
+
+        # ensure the response is valid json
+        data = resp.read()
+        try:
+            json_data = json.loads(ustr(data, encoding="utf-8"))
+        except Exception as e:
+            return False, "JSON parsing failed: {0}".format(ustr(e))
+
+        # ensure all expected fields are present and have a value
+        try:
+            self.check_field(json_data, 'compute')
+            self.check_field(json_data['compute'], 'location')
+            self.check_field(json_data['compute'], 'name')
+            self.check_field(json_data['compute'], 'subscriptionId')
+            self.check_field(json_data['compute'], 'vmSize')
+
+            self.check_field(json_data, 'network')
+            self.check_field(json_data['network'], 'interface')
+            self.check_field(json_data['network']['interface'][0], 'macAddress')
+            self.check_field(json_data['network']['interface'][0], 'ipv4')
+            self.check_field(json_data['network']['interface'][0]['ipv4'], 'ipAddress')
+            self.check_field(json_data['network']['interface'][0]['ipv4']['ipAddress'][0], 'privateIpAddress')
```

Comment:
None of these fields require API version 2018-02-01.  Do you need to verify that API?

Reply:
I don't think it's required, but Hari asked for it. Happy to change.

Reply:
we dont need the new API version , if you want we can use the older one .. agree that these dont need the new version

Reply:
Spoke with Chris, we will leave as is.